### DISCUSSION TITLE
UI overhaul ideas

### DISCUSSION THREAD

**Discussion by nsklaus** [cataclysmbn/Cataclysm-BN#3257] [General]
i was talking about some of my old ideas, like the "in-game codex" in the other thread. i thought it would be good to also bring back the discussion about the general UI for C:BN too, since i saw there are talk on having improvement on that front for this project. to refresh memories, here's what i had in mind with the new panel system originaly:
( and i had a working prototype at the time. now, tell me it doesn't look better that way, when everything is all lined up and displayed cleanly . the idea and code were taken and reworked into what you now have today. i had no saying in the matters. and i view what they have done with it as a far cry of the original intent ).

![sidebar2](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/6810861/5d25ac04-98fe-4fed-806f-809653345664)

![sidebar3](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/6810861/e893bda3-e37f-48d6-9a8e-7536e7bb44b9)

![sidebar4](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/6810861/3e4545f5-188f-4ec9-99d6-49227085b9d1)

the idea was that panels would be toggleable at will. so it would be possible to also have only one, or even zero panel displayed on screen if the player wished it so. it was also about doing _everything_ UI related in the space used by the sidebar. nothing would ever be displayed outside that area. game area would alway be free of any panel showing up on top of it. everything would be displayed as a new, dedicated panel, and be shown in place of some other panel.
so there would have been a `read` panel, a `eat`, a `look around`, a `use`, a `codex`, even `character sheet` and `craft` too (with /filter there would be enough space to show a given recipe and listing needed components comfortably ) ..etc ..
panels needed not to have same height, some might have taken the whole screen height (like the `character sheet` and the `craft` ones probably). one merit is that it segment it in little, self contained,  units. so when working on one panel to implement something or fix a bug, it wouldn't affect the rest of the panel and code would be smaller and easier to look at too.

anyway, that's the subject of this thread. let's see if that might look appealing to the other folks around here or not.

**Comment by chaosvolt** [comment]
Eh, that prototype UI honestly looks even more cursed than DDA's implementation, because of how massively wasteful on vertical space it is due to all the boxes.

I still use classic side to this day, and disable the pixel minimap, because of one big thing: the message log. The current default sidebar plus pixel minimap reduces it down to only 8 lines, and this was if I recall with the more generous window size I tend to favor on my desktop:
![image](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/11582235/52c9ef43-691c-4d3d-aea2-7375478d467b)

This would be even less useful on my laptop. Not seeing a message promptly in this game can frequently mean a YASD.

EDIT: Also gah this discussion had just popped up on my feed why is it actually 3 weeks old hecc
